src/lib/core/TLVReader.cpp

```diff
@@ -324,6 +328,49 @@ CHIP_ERROR TLVReader::Get(CharSpan & v)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR TLVReader::Get(Optional<LocalizedStringIdentifier> & lsid)
+{
+    lsid.ClearValue();
+    VerifyOrReturnError(TLVTypeIsUTF8String(ElementType()), CHIP_ERROR_WRONG_TLV_TYPE);
+
+    const uint8_t * bytes;
+    ReturnErrorOnFailure(GetDataPtr(bytes)); // Does length sanity checks
+
+    uint32_t len = GetLength();
+
+    const uint8_t * infoSeparator1 = reinterpret_cast<const uint8_t *>(memchr(bytes, kUnicodeInformationSeparator1, len));
+    if (infoSeparator1 == nullptr)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    const uint8_t * lsidPtr = infoSeparator1 + 1;
+    len -= static_cast<uint32_t>(lsidPtr - bytes);
+
+    const uint8_t * infoSeparator2 = reinterpret_cast<const uint8_t *>(memchr(lsidPtr, kUnicodeInformationSeparator1, len));
+    if (infoSeparator2 != nullptr)
+    {
+        len = static_cast<uint32_t>(infoSeparator2 - lsidPtr);
+    }
+
+    if (len == 0 || len > kMaxLocalizedStringIdentifierLen)
```

Comment:
`len == 0` is a legit case, but what does it imply if `len > kMaxLocalizedStringIdentifierLen` ?

Shouldn't it return a different error code to indicate there's an error with length of the identifier?

Reply:
Good question! The spec doesn't specify in details what should be the behavior in this case. 
The same question applies for the invalid format when  `Encoding::UppercaseHexToUint16()` fails.
@tennessee-google should the method return error in these cases?

Reply:
Updated the implementation to return error in case of invalid Localized String Identifier value